WalletWasabi/Crypto/Mac.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Text;
+using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto
+{
+	public class MAC : IEquatable<MAC>
+	{
+		private MAC(Scalar t, GroupElement v)
+		{
+			T = CryptoGuard.NotZero(nameof(t), t);
+			V = CryptoGuard.NotNullOrInfinity(nameof(v), v);
+		}
+
+		public Scalar T { get; }
+		public GroupElement V { get; }
+
+		public static bool operator ==(MAC a, MAC b) => a.Equals(b);
+
+		public static bool operator !=(MAC a, MAC b) => !a.Equals(b);
+
+		public bool Equals(MAC? other) =>
+			(this, other) switch
+			{
+				(null, null) => true,
+				(null, { }) => false,
+				({ }, null) => false,
+				({ } a, { } b) when ReferenceEquals(a, b) => true,
+				({ } a, { } b) => (a.T, a.V) == (b.T, b.V)
+			};
```

Comment:
Shouldn't simply `=> (a?.T, a?.V) == (b?.T, b?.V)` just work fine? I think that's the way to do it.

Reply:
Right. Thank you.